aggregator/src/aggregator.rs

```diff
@@ -336,12 +378,25 @@ impl<C: Clock> Aggregator<C> {
         aggregation_job_id: &AggregationJobId,
         req_bytes: &[u8],
         auth_token: Option<AuthenticationToken>,
+        taskprov_header: Option<&[u8]>,
```

Comment:
I don't think it makes sense for either this handler or the aggregate share handler, below, to deal with taskprov. How could we be handling an aggregation job continue or aggregate share request if we didn't first handle an aggregation job init request that provisioned the task?

The [spec](https://datatracker.ietf.org/doc/html/draft-wang-ppm-dap-taskprov-04#name-the-taskprov-extension) says:

>Otherwise, if the Leader does opt-in, it advertises the task to the Helpers during the aggregate protocol ([Section 4.4](https://datatracker.ietf.org/doc/html/draft-ietf-ppm-dap-04#section-4.4) of [[DAP](https://datatracker.ietf.org/doc/html/draft-ietf-ppm-dap-04)]). In particular, it includes the task configuration in an HTTP header *of its first aggregate request for that task*.

(emphasis mine) 

Then the helper section says:
>Finally, the Helper completes the *aggregate initialize* request as usual, deriving the VDAF verification key for the task as described in [Section 3.2](https://datatracker.ietf.org/doc/html/draft-wang-ppm-dap-taskprov-04#vdaf-verify-key).

IMO the leader text is ambiguous and we should ask the taskprov authors to clarify exactly which requests the leader should augment with the taskprov header, but I think the intent here is what I concluded: only the leader's `AggregationJobInitReq` should include the taskprov header.

Reply:
My apologies, I should have linked the pending change that this logic follows https://github.com/wangshan/draft-wang-ppm-dap-taskprov/commit/73859696afcf7a1ddfdcb83de3632e7d1152c327. This PR is based off of the main branch of `draft-wang-ppm-dap-taskprov`, as we went with in https://github.com/divviup/janus/issues/1568#issuecomment-1619156088.

This sentence in particular is what drove me to do it for `continue`.

> The Leader MUST advertise the task to the Helper in every request incident to the task as described in {{definition}}.

And for `share`.

> The Leader MUST advertise the task in every aggregate share request issued to the Helper as described in {{definition}}.

That being said, you are correct that it doesn't really make sense for `share` or `continue` to follow this flow since nothing can be done in these functions without a provisioned task**. It does seem worth it to ask the authors, I'll do so tomorrow.

** I guess you could collect nothing to trigger the taskprov flow and provision a task... perhaps that was their intent?

Reply:
Ahah, this is the intent of "The Leader MUST advertise the task to the Helper in every request incident to the task as described in {{definition}}" https://github.com/wangshan/draft-wang-ppm-dap-taskprov/issues/31, it's so that tasks can be generated on the fly rather than stored in the DB.

Reply:
Ah, I can see how this requirement helps in the face of that design choice. But as you point out, Janus doesn't need to generate tasks on the fly, so while we might still expect to see the `dap-taskprov` header in aggregation job continue and aggregate share requests, we can just ignore it and use the task ID in the request path just like any other request. Regardless of the presence of taskprov, an aggregation job continue req for some (task ID, aggregation job ID) pair is going to fail if it was not preceded by an aggregation job init req.

I know that there's a MUST about checking `TaskConfig` against the task ID in the request path, but I suspect that can be checked higher up in the stack. Or frankly I'd be comfortable with just not checking the `TaskConfig` anywhere but aggregation job init.

Reply:
Ah, looking deeper at implementing this, these requests do need to be concerned with taskprov so that we can use the peer aggregator's auth tokens. The only indication about what auth tokens need to be used comes from the taskprov header.

In essence, this PR using the property "The Leader MUST advertise the task to the Helper in every request incident" to identify the peer aggregator being used per request. This goes away if we agree upon a better authentication or peer aggregator identification mechanism with our peer.

I've noted in slack that some of this complexity comes from supporting multiple peer aggregators. If we accept that there would only be one peer aggregator in one role per janus deployment, then we don't need to parse the taskprov header before authentication.